src/UserSpaceInstrumentation/InstrumentProcess.cpp

```diff
@@ -263,6 +359,39 @@ ErrorMessageOr<std::unique_ptr<InstrumentedProcess>> InstrumentedProcess::Create
                                        process->return_trampoline_address_);
   OUTCOME_TRY(return_trampoline_memory->EnsureMemoryExecutable());
 
+  // Keep track of the threads in the target process before we initialize the user space
+  // instrumentation library.
+  const auto tids_as_vector = orbit_base::GetTidsOfProcess(pid);
+  const absl::flat_hash_set<pid_t> tids_before_injection(tids_as_vector.begin(),
+                                                         tids_as_vector.end());
+
+  // Call initialization code in a new thread.
+  OUTCOME_TRY(auto&& thread_stack_memory, MemoryInTracee::Create(pid, 0, 8ULL * 1024ULL * 1024ULL));
+  const uint64_t top_of_stack = thread_stack_memory->GetAddress() + 8ULL * 1024ULL * 1024ULL;
+  OUTCOME_TRY(auto&& code_memory, MemoryInTracee::Create(pid, 0, 4ULL * 1024ULL));
```

Comment:
Do you need 4 kB? Can you not use the size of what is returned by `MachineCodeForCloneCall`? I guess in the end it's going to be 4 kB anyway because of page size but it seems conceptually wrong or at least unclear here.

Reply:
We only need a few bytes and mmap will round this up to one page. Using code size now as suggested.